test/verify/check-packagekit

```diff
@@ -231,6 +242,249 @@ class TestUpdates(NoSubManCase):
+                        b.wait_text("#delay .pf-c-select__toggle-text", "No delay")
+                        b.click("#shutdown-dialog button:contains('Restart')")
+                        b.switch_to_top()
+                        b.wait_in_text(".curtains-ct h1", "Disconnected")
+
+                        # ensure that rebooting actually worked
+                        m.wait_reboot()
+                        m.start_cockpit()
+                        b.reload()
+                        b.login_and_go("/updates")
+                    else:
+                        # update required a service restart
+                        b.wait_not_present("#reboot-system")
+                        b.click("#choose-service")
+                        b.wait_visible("#restart-services-modal")
+                        b.wait_in_text(".restart-services-modal-body", self.packageName)
+                        b.click("#restart-services-modal button:contains('Restart services')")
+                        b.wait_not_present("#restart-services-modal")
+
+                # check no updates are present
+                b.wait_in_text("#state", "System is up to date")
+
+                # history on "up to date" page should show the recent update (expanded by default)
+                self.testObj.assertHistory("#expanded-content1 > div > ul", [self.packageName])
+
+            def verify_backend(self):
+                if not self.rebootRequired:
+                    # Check the service was actually restarted
+                    newServiceStartTime = m.execute("systemctl show {0}.service --property=ExecMainStartTimestamp".format(self.packageName))
+                    self.testObj.assertGreater(newServiceStartTime, self.serviceStartTime)
+
+                # check no services/processes need reboot or service restart
+                # tracer returns non-zero exit code if any services/processes are affected
+                m.execute("tracer --all --root")
+
+            def cleanup(self):
+                m.execute("rpm -e {0}".format(self.packageName))
+                if self.rebootRequired:
+                    m.execute("rm /etc/tracer/applications.xml")
+
+
+        Tracer(
+            testObj=self,
+            packageName="apple",
+            rebootRequired=True,
+            testStatusCard=False,
+        ).execute()
+
+        Tracer(
+            testObj=self,
+            packageName="cherry",
+            rebootRequired=False,
+            testStatusCard=False,
+        ).execute()
+
+        Tracer(
+            testObj=self,
+            packageName="pear",
+            rebootRequired=True,
+            testStatusCard=True,
+        ).execute()
+
+        Tracer(
+            testObj=self,
+            packageName="banana",
+            rebootRequired=False,
+            testStatusCard=True,
+        ).execute()
+
+        self.allow_restart_journal_messages()
+
+    def testFailServiceRestart(self):
```

Comment:
This test also needs to be skipped on the non-tracer OSes. I suggest to move that OS list to a constant array, and use `skipImage("tracer not supported", *NO_TRACER_OS_LIST)` or so.